### PR TITLE
bugfix: fix duplicated cache key

### PR DESCRIPTION
**Pull request by l81893521**
### Ⅰ. Describe what this PR did
Fix duplicated cache key in table meta cache.

For mysql
```
1.remove the single quote
2.Is case sensitive base on databaseMetaData.supportsMixedCaseIdentifiers().
```

For oracle
```
1.Is case sensitive base on double quote.
(because Oracle does not implement databaseMetaData.supportsMixedCaseIdentifiers())
2.remove the double quote.
```

### Ⅱ. Does this pull request fix one issue?
#1928

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java [hunk 1]
```diff
@@ -65,6 +70,35 @@ public static TableMetaCache getInstance() {
...
+                cacheKey.append(defaultTableName);
+            } else {
+                cacheKey.append(defaultTableName.toLowerCase());
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Could not get supportsMixedCaseIdentifiers in connection metadata, use default cache key", e.getMessage(), e);
+            return cacheKey.append(cacheKey.append(defaultTableName)).toString();
+        }
+
+        return cacheKey.toString();
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: We can return cacheKey.append(defaultTableName).toString() as default.

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java [hunk 2]
```diff
@@ -65,6 +70,35 @@ public static TableMetaCache getInstance() {
...
+        StringBuilder cacheKey = new StringBuilder(dataSourceProxy.getResourceId());
+        cacheKey.append(".");
+        String defaultTableName = tableName.replace("`", "");
+
+        DatabaseMetaData databaseMetaData = null;
+        try {
+            databaseMetaData = dataSourceProxy.getPlainConnection().getMetaData();
+        } catch (SQLException e) {
+            LOGGER.error("Could not get connection, use default cache key", e.getMessage(), e);
+            return cacheKey.append(cacheKey.append(defaultTableName)).toString();
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: Why append itself?Is it cacheKey.append(defaultTableName).toString();

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java [hunk 3]
```diff
@@ -65,6 +70,35 @@ public static TableMetaCache getInstance() {
...
+        try {
+            //prevent duplicated cache key
+            if (databaseMetaData.supportsMixedCaseIdentifiers()) {
+                cacheKey.append(defaultTableName);
+            } else {
+                cacheKey.append(defaultTableName.toLowerCase());
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Could not get supportsMixedCaseIdentifiers in connection metadata, use default cache key", e.getMessage(), e);
+            return cacheKey.append(cacheKey.append(defaultTableName)).toString();
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: Same as above.

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java (hunk 1 above)
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: Thanks. Fixed.

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java (hunk 2 above)
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: Thanks. Fixed.

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java (hunk 1 above)
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: If change, the code would like this
```
StringBuilder cacheKey = new StringBuilder(dataSourceProxy.getResourceId());
        cacheKey.append(".");
        String defaultTableName = tableName.replace("`", "");

        DatabaseMetaData databaseMetaData = null;
        try {
            databaseMetaData = dataSourceProxy.getPlainConnection().getMetaData();
        } catch (SQLException e) {
            LOGGER.error("Could not get connection, use default cache key", e.getMessage(), e);
        }

        try {
            //prevent duplicated cache key
            if (databaseMetaData!= null && !databaseMetaData.supportsMixedCaseIdentifiers()) {
                return cacheKey.append(defaultTableName.toLowerCase()).toString();
            }
        } catch (SQLException e) {
            LOGGER.error("Could not get supportsMixedCaseIdentifiers in connection metadata, use default cache key", e.getMessage(), e);
        }

        return cacheKey.append(defaultTableName).toString();
```

Comment (xingfudeshi) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: OK.

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java [hunk 4]
```diff
@@ -65,6 +70,35 @@ public static TableMetaCache getInstance() {
         return tableMetaCache;
     }
 
+    @Override
+    protected String getCacheKey(DataSourceProxy dataSourceProxy, String tableName) {
+        StringBuilder cacheKey = new StringBuilder(dataSourceProxy.getResourceId());
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: @l81893521 Does tableName need to remove the database name here?

Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java]: Thanks. Fixed.

**Pull request merged**